ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -94,10 +97,27 @@ static PrjFS_Callbacks s_callbacks;
 static dispatch_queue_t s_messageQueueDispatchQueue;
 static dispatch_queue_t s_kernelRequestHandlingConcurrentQueue;
 
-// Map of relative path -> set of pending message IDs for that path, plus mutex to protect it.
-static unordered_map<string, set<uint64_t>> s_PendingRequestMessageIDs;
-static mutex s_PendingRequestMessageMutex;
+struct CaseInsensitiveStringCompare
+{
+    bool operator() (const std::string& lhs, const std::string& rhs) const
+    {
+        return strcasecmp(lhs.c_str(), rhs.c_str()) < 0;
+    }
+};
+
+struct MutexAndUseCount
+{
+    shared_ptr<mutex> mutex;
+    int useCount;
+};
 
+// Map of relative path -> MutexAndUseCount for that path, plus mutex to protect the map itself.
+typedef map<string, MutexAndUseCount, CaseInsensitiveStringCompare> PathToMutexMap;
+PathToMutexMap s_inProgressExpansions;
```

Comment:
Rename to just `s_fileLocks`? Today, this is protecting expansions, but I think it will need to expand to protect all placeholder transitions, e.g. the placeholder->full and hydrated->empty transitions as well.